### PR TITLE
[MRG] gp_minimize should not use default estimator when a custom one is provided

### PR DESCRIPTION
**Pull request by stefanocereda**
I think that gp_minimize should use the given optimizer, if it exists

### CODE REVIEW

**Comment by betatim** [comment]
Indeed! Could you add a test so this bug doesn't come back?

**Comment by stefanocereda** [comment]
I've added a test, let'me know if it's ok as it's my first one

File: skopt/tests/test_gp_opt.py [hunk 1]
```diff
@@ -71,3 +73,17 @@ def test_gpr_default():
     """Smoke test that gp_minimize does not fail for default values."""
     gp_minimize(branin, ((-5.0, 10.0), (0.0, 15.0)), n_random_starts=1,
                 n_calls=2)
+
+
+@pytest.mark.fast_test
+def test_use_given_estimator():
+    """ Check that gp_minimize use the given estimator. """
+    domain = [(1.0, 2.0), (3.0, 4.0)]
+    space = normalize_dimensions(domain)
```
Comment (betatim) [inline skopt/tests/test_gp_opt.py]: do we need this line?

File: skopt/tests/test_gp_opt.py [hunk 2]
```diff
@@ -71,3 +73,17 @@ def test_gpr_default():
     """Smoke test that gp_minimize does not fail for default values."""
     gp_minimize(branin, ((-5.0, 10.0), (0.0, 15.0)), n_random_starts=1,
                 n_calls=2)
+
+
+@pytest.mark.fast_test
+def test_use_given_estimator():
+    """ Check that gp_minimize use the given estimator. """
```
Comment (betatim) [inline skopt/tests/test_gp_opt.py]: I'd change this to `# Test that gp_minimize does not use default estimator if one is passed in explicitly.`

File: skopt/tests/test_gp_opt.py (hunk 1 above)
Comment (stefanocereda) [reply skopt/tests/test_gp_opt.py]: I have to give a Space to cook_estimator so I think it's needed

**Comment by betatim** [comment]
If you rebase your branch on the latest master the CircleCI failure should be fixed as well.

File: skopt/tests/test_gp_opt.py (hunk 1 above)
Comment (betatim) [reply skopt/tests/test_gp_opt.py]: `cook_estimator("GP", [(1.0, 2.0), (3.0, 4.0)], noise=1234)` also works no? To save the call to `normalize_dimensions`

**Comment by betatim** [comment]
Thanks for catching this bug! :clinking_glasses:

**Pull request merged**